examples/bin/start-druid-main

```diff
@@ -0,0 +1,436 @@
+#!/usr/bin/env python3
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import sys
+import os
+import psutil
+import pathlib
+import multiprocessing
+import argparse
+
+QUICKSTART_ROOT_CONFIG_PATH = "conf/druid/single-server/quickstart"
+
+MEMORY_GIGABYTES_IDENTIFIER = "g"
+MEMORY_MEGABYTES_IDENTIFIER = "m"
+SERVICE_SEPARATOR = ","
+
+MM_TASK_JAVAOPTS_ARRAY = ["-server", "-Duser.timezone=UTC","-Dfile.encoding=UTF-8","-XX:+ExitOnOutOfMemoryError","-Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager"]
+MM_TASK_MEMORY_TYPE_LOW = "low"
+MM_TASK_MEMORY_TYPE_HIGH = "high"
+MM_TASK_MEM_MAP = {
+    MM_TASK_MEMORY_TYPE_LOW: ["-Xms256m", "-Xmx256m", "-XX:MaxDirectMemorySize=256g"],
+    MM_TASK_MEMORY_TYPE_HIGH: ["-Xms1g", "-Xmx1g", "-XX:MaxDirectMemorySize=1g"]
+}
+
+MM_TASK_JAVAOPTS_PROP = "-Ddruid.indexer.runner.javaOptsArray"
+MM_TASK_WORKER_CAPACITY_PROP = "-Ddruid.worker.capacity"
+
+BROKER = "broker"
+ROUTER = "router"
+COORDINATOR = "coordinator-overlord"
+HISTORICAL = "historical"
+MM = "middleManager"
+MM_TASK = "middleManager-task"
+MM_TASK_COUNT = "task-count"
+
+DEFAULT_SERVICES = [
+    BROKER,
+    ROUTER,
+    COORDINATOR,
+    HISTORICAL,
+    MM
+]
+
+SERVICE_MEMORY_DISTRIBUTION_WEIGHT = {
+    MM: 0.5,
+    ROUTER: 1,
+    COORDINATOR: 15,
+    BROKER: 23,
+    HISTORICAL: 40,
+    MM_TASK: 15
```

Comment:
Is this weight for a single task or all tasks put together?

If it's for all tasks, having a constant value for this might not be the best. Larger configurations would have several tasks whereas smaller configurations would have fewer tasks. Thus the weights would change.

If it's for a single task, the function `distribute_memory_over_services` should take into account the number of tasks.

Reply:
> Is this weight for a single task or all tasks put together? 

The weight is for all the tasks put together. 

> If it's for all tasks, having a constant value for this might not be the best. Larger configurations would have several tasks whereas smaller configurations would have fewer tasks. Thus the weights would change.

The memory allocated for all tasks would increase proportionately for larger configuration. Since task memory is fixed, it would automatically lead to higher task count. 
I didn't get the part about weights changing? do you mean to say we should allocated bigger chunk of memory in larger profiles?